internal/connector/connector.go

```diff
@@ -52,6 +51,15 @@ type ListenerOptions struct {
 	Metrics string
 }
 
+// connector stores all the dependencies for the connector operations.
+type connector struct {
```

Comment:
I'm never sure what to call these, so I went with the same pattern we used for `Server` and called it `connector`. As the code changes maybe we'll find a better name for this, or find ways to split it into other types.

I'm doing this now to allow us to split up `syncWithServer` and removing the need to pass around 5+ args to the functions.